index.js

```
@@ -1,301 +1,319 @@
+ * @extends {fs.WriteStream}
+ */
+class RotateStream extends fs.WriteStream {
+    /**
+     * Given an local path and options initialize listeners,
+     * and return Stream instance ready to be writen on
+     * @param path
+     * @param options
+     */
+    constructor(path, options) {
+        options = Object.assign({}, DEFAULT_OPTIONS, options)
+        if (options.flags.indexOf('a') === -1) {
+            throw new Error('RotateStream must be opened in append mode')
+        }
+        super(path, options)
+        this.on('open', () => {
+            fs.fstat(this.fd, (err, stats) => {
+                if (err) {
+                    this.emit('error', err);
+                } else {
+                    this.ino = stats.ino;
+                }
+            })
+        })
 
-util.inherits( RotateStream, fs.WriteStream );
-function RotateStream( path, options ) {
-    options || ( options = {} );
+        this.compress = options.gzip
+        this.size = isNaN(options.size) ? bytes(options.size) : options.size
+        this.suffix = options.suffix
+        if (options.upload) {
+            if (!aws) {
+                throw new Error('The aws-sdk module isn\'t installed')
+            }
 
-    if ( !options.flags ) {
-        options.flags = 'a';
+            this.toUpload = true
+            this._s3 = _getS3(options.upload)
+        }
     }
 
-    if ( !options.size ) {
-        options.size = '5gb';
-    }
+    /**
+     * @private
+     */
+    _openAsync() {
+        var self = this
 
-    if ( options.flags.indexOf( 'a' ) == -1 ) {
-        throw new Error( 'RotateStream must be opened in append mode' );
+        return fs.openAsync(this.path, this.flags, this.mode)
+            .then((fd) => {
+                self.fd = fd
+                self.emit('open', fd)
+            })
+            .catch((err) => {
+                if (self.autoClose) {
+                    self.destroy()
+                }
+                self.emit('error', err)
```

You're swallowing the error here - by just emitting the error but not re-raising it. This means that the caller will never see that an error has occurred outside of the `error` event. Is this intentional?

---

Again I have fixed this, now errors are not suppressed unless there is a reason for it